### PR TITLE
add initial security policy resources

### PR DESCRIPTION
**Pull request by bobcallaway**
Signed-off-by: Bob Callaway <bcallaway@google.com>

### CODE REVIEW

**Review by k4leung4 — APPROVED** [review]

**Review by vaikas — DISMISSED** [review]

File: terraform/gcp/modules/rekor/security_policy.tf [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+ */
+
+// Enable required services for this module
+resource "google_compute_security_policy" "rekor" {
+  name    = "rekor-service-security-policy"
+  project = var.project_id
+  type    = "CLOUD_ARMOR"
+
+  rule {
+    action   = "deny(502)"
```
Comment (var-sdk) [inline terraform/gcp/modules/rekor/security_policy.tf:24]: Any reason why not 413?

Comment (bobcallaway) [reply terraform/gcp/modules/rekor/security_policy.tf:24]: Unfortunately the only options allowed by GCP are: 

deny(403), deny(404), deny(429) deny(502)

**Review by var-sdk — DISMISSED** [review]
Sorry for the slow review, I went down the rabbit hole of trying to understand how these fit together with GKE and our helm charts. The factoring across repos is hard to grok but I don't have any better suggestions.

File: terraform/gcp/modules/rekor/security_policy.tf [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    priority = "1"
+
+    match {
+      expr {
+        expression = "int(request.headers['content-length']) > 134217728"
+      }
+    }
+    description = "Block all incoming requests > 128MB"
```
Comment (haydentherapper) [inline terraform/gcp/modules/rekor/security_policy.tf:32]: Will this affect any current callers sending alpine packages? Or is this limit already in place?

File: terraform/gcp/modules/fulcio/security_policy.tf [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Enable required services for this module
+resource "google_compute_security_policy" "fulcio" {
+  name    = "fulcio-service-security-policy"
+  project = var.project_id
+  type    = "CLOUD_ARMOR"
+
```
Comment (haydentherapper) [inline terraform/gcp/modules/fulcio/security_policy.tf:22]: Can we add a content length restriction here too? Someone could construct a massive CSR

File: terraform/gcp/modules/rekor/security_policy.tf (hunk 2 above)
Comment (bobcallaway) [reply terraform/gcp/modules/rekor/security_policy.tf:32]: same as https://github.com/sigstore/public-good-instance/blob/5c50dcf1b55537685609453b751507d136a05f29/argocd/utilities/templates/scaffold.yaml#L250

**Review by haydentherapper — APPROVED** [review]

**Pull request merged**